avatica/core/src/main/java/org/apache/calcite/avatica/AvaticaConnection.java

```diff
@@ -664,6 +665,13 @@ private int unbox(Integer i, int defaultValue) {
     return meta.connectionSync(handle, new ConnectionPropertiesImpl());
   }
 
+  /** Returns or creates a slot whose state can be changed to cancel a
+   * statement. */
+  public AtomicBoolean getCancelFlag(Meta.StatementHandle h)
+      throws NoSuchStatementException {
+    return new AtomicBoolean();
```

Comment:
Isn't this a worry if, by some means, we call `connection.getCancelFlag(h)` for the same `h` multiple times and end up with multiple instances? Can we initialize an `AtomicBoolean` on the `StatementHandle` itself?